### PR TITLE
Support Chronos in TSFM Inference Service

### PR DESCRIPTION
**Pull request by gganapavarapu**
This PR is to enable Chronos in TSFM inference service.

### CODE REVIEW

**Comment by ssiegel95** [comment]
@gganapavarapu let's redirect the target branch to `services_chronos_support`

**Comment by ssiegel95** [comment]
@gganapavarapu have you run `make style` on your code? it seems some of the deltas are formatting in nature.

**Comment by gganapavarapu** [comment]
> @gganapavarapu have you run `make style` on your code? it seems some of the deltas are formatting in nature.

Yes, I have run it. It is not reverting some of the style changes.

**Comment by ssiegel95** [comment]
Sorry to ask this again but for some reason the github actions is not happening on this PR request despite me specifying this branch in the config file. Could you target main again since @wgifford has recently merged #171 ? Thanks.

File: services/inference/Makefile [hunk 1]
```diff
@@ -62,6 +62,8 @@ install_dev: boilerplate
 clone_models:
 	git lfs install || true
 	git clone https://huggingface.co/ibm/test-tsfm mytest-tsfm || true
+	git clone https://huggingface.co/amazon/chronos-t5-small mytest-tsfm/chronos-t5-small || true
```
Comment (wgifford) [inline services/inference/Makefile:65]: Instead, can we update `test-tsfm` to have a `chronos-t5-small` folder containing the tsfm-config already?

File: services/inference/tsfm_config.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+    "service_handler_module_path": "tsfminference.chronos_service_handler",
+    "service_handler_class_name": "ChronosForecastingHandler",
+    "maximum_prediction_length": 64,
+    "minimum_context_length": 256,
```
Comment (wgifford) [inline services/inference/tsfm_config.json:5]: Are these parameters correct? Does chronos have min/max context?

File: services/inference/tsfminference/chronos_service_handler.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+    """
+
+    def __init__(
+        self,
+        model_id: str,
+        model_path: Union[str, Path],
+        handler_config: TSFMConfig,
+    ):
+        # !!! Double check which init
```
Comment (wgifford) [inline services/inference/tsfminference/chronos_service_handler.py:41]: Let's remove this comment

File: services/inference/tsfminference/chronos_service_handler.py [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+        Returns:
+            pd.DataFrame: The forecasts produced by the model.
+        """
+
+        target_columns = schema.target_columns or data.columns.tolist()
+        prediction_length = (
+            parameters.prediction_length or self.chronos_config["prediction_length"]
+        )
+
+        num_samples = self.chronos_config["num_samples"]
```
Comment (wgifford) [inline services/inference/tsfminference/chronos_service_handler.py:105]: Can these be part of parameters?

File: services/inference/Makefile (hunk 1 above)
Comment (gganapavarapu) [reply services/inference/Makefile:65]: @wgifford I am trying to push tsfm_config.json to test_tsfm/chronos-t5-small/tsfm_config.json and create a PR. Could you give me access to this repository?

Having duplicated copy of chronos model in this repository may not be ideal. I don't see a way to use submodules given we need to add tsfm_config.json as well. 

We can handle this in makefile: 
1. Persist tsfm_config files for all models in a separate directory: tsfm_config_files/chronos/tsfm_config.json.
2. Clone chronos model repository -> we can do submodules too.
3. Copy tsfm_config_files/chronos/tsfm_config.json to chronos model path.

What do you think?

File: services/inference/tsfminference/chronos_service_handler.py (hunk 4 above)
Comment (gganapavarapu) [reply services/inference/tsfminference/chronos_service_handler.py:104]: Current version of service APIs do no allow model specific run time parameters. I have created #204  to track.

**Comment by gganapavarapu** [comment]
> @wgifford any insight into why the tests are not kicking off on this PR?

It seems I need to approve the workflow actions under Actions tab. That triggered the approval request from maintainers.

**Comment by gganapavarapu** [comment]
Updated the PR to enable only `chronos-t5-tiny` for now.

**Comment by ssiegel95** [comment]
@gganapavarapu  please run poetry lock on a new clone of your repo **from a linux machine** and push changes. it's sometimes hard to resolve this issue via merging.

**Comment by gganapavarapu** [comment]
> @gganapavarapu please run poetry lock on a new clone of your repo **from a linux machine** and push changes. it's sometimes hard to resolve this issue via merging.

I don't have a linux machine at the moment. Let me try to get one. If this something you can run quickly, it will greatly help!

**Comment by ssiegel95** [comment]
> > @gganapavarapu please run poetry lock on a new clone of your repo **from a linux machine** and push changes. it's sometimes hard to resolve this issue via merging.
> 
> I don't have a linux machine at the moment. Let me try to get one. If this something you can run quickly, it will greatly help!

Tough for me because I can only affect change on your branch for a PR process which I tried this morning and that doesn't seem to be working.

**Comment by ssiegel95** [comment]
@gganapavarapu if you add me as a collaborator to your fork then I think I could do it more easily.

**Comment by ssiegel95** [comment]
@gganapavarapu it'd probably be okay to run from your mac as long as you did a `git checkout pyproject.toml` first to make you sure you don't have any local change related to the missing torch cpu whl for Mac OS.

**Comment by gganapavarapu** [comment]
> @gganapavarapu it'd probably be okay to run from your mac as long as you did a `git checkout pyproject.toml` first to make you sure you don't have any local change related to the missing torch cpu whl for Mac OS.

I have seen mac related files generated in poetry.lock like below and so, did not push it.
```
{file = "Brotli-1.1.0-cp313-cp313-macosx_10_13_universal2.whl", hash = "sha256:8bf32b98b75c13ec7cf774164172683d6e7891088f6316e54425fde1efc276d5"},
+    {file = "Brotli-1.1.0-cp313-cp313-macosx_10_13_x86_64.whl", hash = "sha256:7bc37c4d6b87fb1017ea28c9508b36bbcb0c3d18b4260fcdf08b200c74a6aee8"},
```

**Comment by ssiegel95** [comment]
@gganapavarapu at least now we're down to actual code errors :-).

**Comment by gganapavarapu** [comment]
Updated target branch to `new_model_integrations` to track all model additions.

**Comment by ssiegel95** [comment]
@gganapavarapu I keep trying to resolve this but it doesn't stick. We want to go with `branches: [ "main", "new_model_integrations" ]`

```
<<<<<<< service_abstraction_chronos
    branches: [ "main", "services_chronos_support", "destiny" ]
=======
    branches: [ "main", "new_model_integrations" ]
>>>>>>> new_model_integrations
```

File: services/inference/Makefile (hunk 1 above)
Comment (gganapavarapu) [reply services/inference/Makefile]: We moved this file to test-tsfm repository with this content:
```
{
    "service_handler_module_path": "tsfminference.chronos_service_handler",
    "service_handler_class_name": "ChronosForecastingHandler",
    "minimum_context_length": 512,
    "module_path": "chronos",
    "model_type": "chronos-t5-tiny",
    "model_config_name": "T5Config",
    "model_class_name": "ChronosPipeline"
}
```

Thank you @wgifford

**Pull request merged**